extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamInfoItemExtractor.java

```diff
@@ -72,10 +72,14 @@ public StreamType getStreamType() {
         final JsonArray thumbnailOverlays = videoInfo.getArray("thumbnailOverlays");
         for (final Object object : thumbnailOverlays) {
             final JsonObject thumbnailOverlay = (JsonObject) object;
-            if (thumbnailOverlay.has("thumbnailOverlayNowPlayingRenderer")
-                    || thumbnailOverlay.getObject("thumbnailOverlayTimeStatusRenderer")
-                    .getString("style", EMPTY_STRING).equalsIgnoreCase("LIVE")) {
-                return cachedStreamType = StreamType.LIVE_STREAM;
+            if (thumbnailOverlay.has("thumbnailOverlayNowPlayingRenderer")) {
+                final String overlayText = thumbnailOverlay
+                        .getObject("thumbnailOverlayTimeStatusRenderer")
+                        .getString("style", EMPTY_STRING);
+                if (overlayText.equalsIgnoreCase("LIVE")
```

Comment:
When you reverse the comparing here to ``"LIVE".equalsIgnoreCase(overlayText)`` (and also in the next line) the code will be NPE save 😄

Reply:
`getString("style", EMPTY_STRING)` returns either a string at the key "style" or an empty String, but not null